seed/php-sdk/imdb/clientName/src/Core/Client/RetryMiddleware.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace Seed\Core\Client;
+
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\Promise as P;
+use GuzzleHttp\Promise\PromiseInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Throwable;
+
+class RetryMiddleware
+{
+    private const RETRY_STATUS_CODES = [408, 429];
+
+    /**
+     * @var callable(RequestInterface, array): PromiseInterface
+     * @phpstan-ignore missingType.iterableValue
+     */
+    private $nextHandler;
+
+    /**
+     * @var array{
+     *     maxRetries: int,
+     *     baseDelay: int,
+     * }
+     */
+    private array $options;
+
+    /**
+     * @param callable $nextHandler
+     * @param array{
+     *     maxRetries?: int,
+     *  }|null $options
+     */
+    public function __construct(
+        callable $nextHandler,
+        ?array   $options = null,
+    ) {
+        $this->nextHandler = $nextHandler;
+        $this->options = array_merge([
+            'maxRetries' => 2,
+            'baseDelay' => 1000,
```

Comment:
Nice - thoughts on defining this as an un-exported constant at the top of the file? That way it's easy to see all the default retry configurations that we're using. Something like:

 ```php
const DEFAULT_RETRY_CONFIG = [
    'maxRetries' => 2,
    'baseDelay' => 1000
];
```

Reply:
Great call

Reply:
Done